From f8008136243ccc40affea8809611a350e5a1997e Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sun, 17 Sep 2017 23:04:45 +0200
Subject: [PATCH 1/1] efi_loader: ProtocolsPerHandle argument
 ProtocolBufferCount

The argument ProtocolBufferCount of ProtocolsPerHandle is defined
as UINTN in the UEFI spec. So we should use size_t and not
unsigned long in efi_protocols_per_handle.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 include/efi_api.h             | 2 +-
 lib/efi_loader/efi_boottime.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/efi_api.h b/include/efi_api.h
index a292cf7d76..e35270f10a 100644
--- a/include/efi_api.h
+++ b/include/efi_api.h
@@ -142,7 +142,7 @@ struct efi_boot_services {
 			unsigned long *entry_count);
 	efi_status_t (EFIAPI *protocols_per_handle)(efi_handle_t handle,
 			efi_guid_t ***protocol_buffer,
-			unsigned long *protocols_buffer_count);
+			size_t *protocols_buffer_count);
 	efi_status_t (EFIAPI *locate_handle_buffer) (
 			enum efi_locate_search_type search_type,
 			const efi_guid_t *protocol, void *search_key,
diff --git a/lib/efi_loader/efi_boottime.c b/lib/efi_loader/efi_boottime.c
index e62b4c342d..f795a57787 100644
--- a/lib/efi_loader/efi_boottime.c
+++ b/lib/efi_loader/efi_boottime.c
@@ -1628,7 +1628,7 @@ static efi_status_t EFIAPI efi_open_protocol_information(efi_handle_t handle,
  */
 static efi_status_t EFIAPI efi_protocols_per_handle(void *handle,
 			efi_guid_t ***protocol_buffer,
-			unsigned long *protocol_buffer_count)
+			size_t *protocol_buffer_count)
 {
 	unsigned long buffer_size;
 	struct efi_object *efiobj;
-- 
2.11.0

